transforms/code/ingest_2_parquet/ray/src/ingest_2_parquet_transform.py

```diff
@@ -0,0 +1,271 @@
+                                    "hash": TransformUtils.str_to_hash(content_string),
+                                    "size": len(content_string),
+                                    "date_acquired": datetime.now().isoformat(),
+                                }
+                                if self.detect_programming_lang:
+                                    lang = self._get_lang_from_ext(ext)
+                                    row_data["programming_language"] = lang
+                                data.append(row_data)
+                                number_of_rows += 1
+                            else:
+                                logger.warning(f"file {member.filename} is empty. content {content_string}, skipping")
+                        except Exception as e:
+                            logger.warning(f"Exception {str(e)} processing file {member.filename}, skipping")
+        table = pa.Table.from_pylist(data)
+        return [(TransformUtils.convert_arrow_to_binary(table=table), ".parquet")], {"number of rows": number_of_rows}
+
+
+class IngestToParquetRuntime(DefaultRayTransformRuntime):
+    """
+    Ingest to Parquet runtime support
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        """
+        Create filter runtime
+        :param params: parameters, that should include
+            ingest_supported_langs_file_key: supported languages file
+            ingest_detect_programming_lang_key: whether to detect programming language
+            ingest_domain_key: domain
+            ingest_snapshot_key: snapshot
+        """
+        super().__init__(params)
+
+    def get_transform_config(
+            self,
+            data_access_factory: DataAccessFactoryBase,
+            statistics: ActorHandle,
+            files: list[str],
+    ) -> dict[str, Any]:
+        """
+        Set environment for filter execution
+        :param data_access_factory - data access factory
+        :param statistics - reference to the statistics object
+        :param files - list of files to process
+        :return: dictionary of filter init params
+        """
+        lang_file = self.params.get(ingest_supported_langs_file_key, None)
+        if lang_file is None:
+            raise RuntimeError(f"Missing configuration key {ingest_supported_langs_file_key}")
+        lang_data_access_factory = self.params.get(ingest_data_factory_key, None)
+        if lang_data_access_factory is None:
+            raise RuntimeError(f"Missing configuration key {ingest_data_factory_key}")
+        lang_dict = _get_supported_languages(
+            lang_file=lang_file,
+            data_access=lang_data_access_factory.create_data_access(),
+        )
+        lang_refs = ray.put(lang_dict)
+        logger.debug(f"Placed language list into Ray object storage under reference{lang_refs}")
+        return {ingest_supported_languages: lang_refs} | self.params
+
+
+class IngestToParquetTransformConfiguration(TransformConfiguration):
```

Comment:
Think we should be consistent on the naming of classes and files like the other transforms.  Pick `ingesttoparquet` or `ingest2parquet` or `ingest_2_parquet` or `ingest_to_parquet`, but lets be consistent.

Reply:
File names and class names do not have to be the same. File names are dependent on directory name in the make files

Reply:
consistency in conventions will help everyone in the end.

Also, currently this is for code, but i think the goal is to have this be more generic.  As such, it belongs in the universal tree not the code tree.  Yes, in anticipation of future extensions and not having to move it later.